src/software/ai/hl/stp/tactic/receiver/receiver_fsm.cpp

```diff
@@ -156,37 +158,37 @@ void ReceiverFSM::updateReceive(const Update& event)
 
 void ReceiverFSM::adjustReceive(const Update& event)
 {
-    auto ball      = event.common.world_ptr->ball();
-    auto robot_pos = event.common.robot.position();
+    const Ball& ball   = event.common.world_ptr->ball();
+    const Robot& robot = event.common.robot;
 
-    if ((ball.position() - robot_pos).length() >
-        BALL_TO_FRONT_OF_ROBOT_DISTANCE_WHEN_DRIBBLING)
-    {
-        Point ball_receive_pos = ball.position();
-
-        if (ball.velocity().length() > MIN_PASS_START_SPEED)
-        {
-            ball_receive_pos = closestPoint(
-                robot_pos, Line(ball.position(), ball.position() + ball.velocity()));
-        }
-
-        Angle ball_receive_orientation = (ball.position() - robot_pos).orientation();
+    auto face_ball_orientation = (ball.position() - robot.position()).orientation();
 
-        event.common.set_primitive(std::make_unique<MovePrimitive>(
-            event.common.robot, ball_receive_pos, ball_receive_orientation,
-            TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
-            TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE,
-            TbotsProto::DribblerMode::MAX_FORCE, TbotsProto::BallCollisionType::ALLOW,
-            AutoChipOrKick{AutoChipOrKickMode::OFF, 0}));
+    Point intercept_position;
+    if (strayPass(event) || ball.velocity().length() <= MIN_STRAY_PASS_SPEED)
```

Comment:
why do we need to this for a stray pass and not use the `else` conditional

Reply:
The interception algorithm used for stray passes and stagnant balls is the same as the one in `DribbleSkillFSM`. This algorithm draws a line between the robot and the ball and tries to find the optimal point on that line to intercept the ball. Hence, it can "chase" the ball if it goes astray. The `closestPoint` interception algorithm used in the else conditional prevents chasing the ball, but if the ball goes past the robot then the robot will not go after it.